src/backend/in_memory_record.rs

```diff
@@ -114,7 +114,7 @@ impl<Datum: Clone + PartialEq + Debug, Redeemer: Clone + Eq + PartialEq + Debug
             let index = my_outputs
                 .iter()
                 .position(|(_, x)| x == tx_i)
-                .ok_or(format!("Input: {:?} doesn't exist", &tx_i))?;
+                .ok_or(Error::TxORecord(TxORecordError::InputDoesNotExist(tx_i.id().to_string())))?;
```

Comment:
We should change the return type on this trait so `issue` returns a `TxORecordError` instead of a core error.

You could even define a `Result` for that error type and import it as `TxORecordResult` or whatever.

Reply:
what module should I define `TxORecordResult` in `backend`?

Reply:
Prolly just pull `TxORecord` out into its own mod and do it next to that.

Reply:
Also since it returns a `Result<()>` I don't think there's anything wrong with it returning a `TxORecordError`

Reply:
I'm not sure I follow.

I think `TxORecordError` should be part of the `TxORecord` interface. Are you thinking something different?